### ISSUE TITLE
Rename C# Script resets script content

### ISSUE THREAD

**Issue by trommlbomml** [godotengine/godot#41905]
**Godot version:**
3.2.3 RC6 Mono

**OS/device including version:**
Ubuntu 20.04
AMD Ryzen 5 3600
Radeon RX550

**Issue description:**
When I have a C# script and rename it the content gets lost and is resetted to the template content. It was really annyoing as I had some more changes and thought it is save to rename a script file and all of them got lost.

**Steps to reproduce:**
1. Have a Godot project open with C# Support 
2. Create a C# Script called "Test.cs", create a field `private int _test;`
3. Go to FileSystem dock and rename "Test.cs" to "Test2.cs".
4. private field is removed.

**Comment by Calinou** [comment]
Did you save the script by pressing <kbd>Ctrl + Alt + S</kbd> (or <kbd>Ctrl + S</kbd<) before renaming it?

**Comment by trommlbomml** [comment]
You are right - even saving in vscode as external editor it gets preserved, so maybe this should an improvement to detect if the script is currently changed by the external editor. Maybe not an easy task.
